Add `no-thenable` rule (#1616)
sindresorhus/eslint-plugin-unicorn: docs/rules/no-thenable.md

# Disallow `then` property

✅ *This rule is part of the [recommended](https://github.com/sindresorhus/eslint-plugin-unicorn#recommended-config) config.*

> If an object is defined as "thenable", once it's accidentally used in an await expression, it may cause problems.

Reviewer: It should describe what kind of problem.

> If an object is defined as "thenable", once it's accidentally used in an await expression, it may cause problems.

Reviewer 2: Any docs to link to about this for context?

Author: I can't find any link for this.

> If a module have exported `then`, dynamic `import()` may won't works as expected.

Reviewer: ```suggestion
If a module has an export named `then`, dynamic `import()` may not work as expected.
```

This also should describe how it will not work.

# Disallow `then` property

✅ *This rule is part of the [recommended](https://github.com/sindresorhus/eslint-plugin-unicorn#recommended-config) config.*

If an object is defined as "thenable", once it's accidentally used in an await expression, it may cause problems.

```js
const foo = {
	unicorn: 1,
	then() {},
};
const {unicorn} = await foo;

> console.log('after'); //<- This will nerve execute

Author: ```suggestion
console.log('after'); //<- This will never execute
```
